api/clients/v2/coretypes/encoded_payload.go

```diff
@@ -121,27 +121,26 @@ func (ep *EncodedPayload) decodeHeader() (uint32, error) {
 	default:
 		return 0, fmt.Errorf("unknown encoded payload header version: %x", ep.bytes[1])
 	}
+
+	for _, b := range ep.bytes[6:codec.EncodedPayloadHeaderLenBytes] {
+		if b != 0x00 {
+			return 0, fmt.Errorf("padding in encoded payload header must be 0x00: %x", b)
+		}
+	}
+
 	return payloadLength, nil
 }
 
 func (ep *EncodedPayload) decodePayload(payloadLen uint32) ([]byte, error) {
 	body := ep.bytes[codec.EncodedPayloadHeaderLenBytes:]
 	// Decode the body by removing internal 0 byte padding (0x00 initial byte for every 32 byte chunk)
 	// The decodedBody should contain the payload bytes + potentially some external padding bytes.
-	decodedBody, err := codec.CheckAndRemoveInternalPadding(body)
+	decodedPayload, err := checkAndRemoveInternalPadding(body, payloadLen)
 	if err != nil {
 		return nil, fmt.Errorf("remove internal padding: %w", err)
 	}
 
-	// data length is checked when constructing an encoded payload. If this error is encountered, that means there
-	// must be a flaw in the logic at construction time (or someone was bad and didn't use the proper construction methods)
-	if uint32(len(decodedBody)) < payloadLen {
-		return nil, fmt.Errorf(
```

Comment:
If you don't want to return an error here, I think it still makes sense to log this occurrence as a sanity check